### PR TITLE
feat(cli): add dashboard snapshot command with since filter and KPI tables (#119)

### LINKED ISSUE #119: Post-alpha: add dashboard CLI snapshot for operator visibility
Priority: Medium\n\nContext:\nAlpha delivered report/tail primitives, but operators still need a single command that summarizes current spend posture at a glance without manually combining multiple commands.\n\nProblem:\n- There is no dedicated dashboard command in CLI.\n- Users must run separate report commands to see key totals and top contributors.\n\nScope (MVP):\n- Add dashboard command (non-interactive snapshot).\n- Show header KPIs: total requests, total input/output tokens, total cost (USD).\n- Show top groups by project and by model (same query semantics as report summary).\n- Support --since for rolling windows (same parser as report summary).\n- Keep output deterministic and script-friendly (table text).\n\nAcceptance Criteria:\n- dashboard works against current SQLite schema with no migrations required.\n- --since filtering behaves consistently with existing report summary semantics.\n- Unit tests cover rendering and aggregation behavior for non-empty and empty datasets.\n- Workspace checks remain green: fmt, test, check, clippy.\n\nReferences:\n- PennyPrompt-v2.md (Post-Alpha objective: dashboard usability)

### PR DESCRIPTION
**Pull request by manuelpenazuniga**
## Summary
- add `pennyprompt dashboard` command for operator snapshot visibility
- reuse `--since` parsing semantics from report summary
- render KPI totals (requests, tokens, total cost) plus top project/model sections
- add unit tests for empty and bounded-top rendering behavior

## Validation
- cargo fmt --all
- cargo test -p penny-cli --locked
- cargo check --workspace --locked
- cargo clippy --workspace --all-targets --locked -- -D warnings

Closes #119

### CODE REVIEW

File: crates/penny-cli/src/main.rs [hunk 1]
```diff
@@ -2196,6 +2211,121 @@ fn print_summary(
...
+}
+
+fn compute_dashboard_totals(rows: &[SummaryRow]) -> DashboardTotals {
+    DashboardTotals {
+        request_count: rows.iter().map(|row| row.request_count).sum(),
+        input_tokens: rows.iter().map(|row| row.input_tokens).sum(),
+        output_tokens: rows.iter().map(|row| row.output_tokens).sum(),
+        total_cost_usd: rows.iter().map(|row| row.total_cost_usd).sum(),
+    }
+}
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/penny-cli/src/main.rs:2236]: Ok
